bundles/org.openhab.binding.freeathomesystem/src/main/java/org/openhab/binding/freeathomesystem/internal/handler/FreeAtHomeBridgeHandler.java

```diff
@@ -0,0 +1,872 @@
+            }
+        }
+
+        WebSocketClient localWebSocketClient = websocketClient;
+
+        if (localWebSocketClient != null) {
+            try {
+                localWebSocketClient.stop();
+            } catch (Exception e2) {
+                logger.debug("Error by closing Websocket connection [{}]", e2.getMessage());
+            }
+        }
+    }
+
+    /**
+     * Method to open the websocket connection
+     */
+    public boolean openWebSocketConnection() {
+        boolean ret = false;
+
+        QueuedThreadPool localThreadPool = jettyThreadPool;
+
+        if (localThreadPool == null) {
+            jettyThreadPool = new QueuedThreadPool();
+
+            localThreadPool = jettyThreadPool;
+
+            if (localThreadPool != null) {
+                localThreadPool.setName(FreeAtHomeBridgeHandler.class.getSimpleName());
+                localThreadPool.setDaemon(true);
+                localThreadPool.setStopTimeout(0);
+
+                ret = true;
+            }
+        }
+
+        WebSocketClient localWebSocketClient = websocketClient;
+
+        if (localWebSocketClient == null) {
+            websocketClient = new WebSocketClient(httpClient);
+
+            localWebSocketClient = websocketClient;
+
+            if (localWebSocketClient != null) {
+                localWebSocketClient.setExecutor(jettyThreadPool);
+
+                socketMonitor.start();
+
+                ret = true;
+            } else {
+                ret = false;
+            }
+        } else {
+            ret = true;
+        }
+
+        return ret;
+    }
+
+    /**
+     * Method to initialize the bridge
+     *
+     *
+     */
+    @Override
+    public void initialize() {
+        scheduler.execute(() -> {
+            boolean thingReachable = true;
+
+            updateStatus(ThingStatus.UNKNOWN);
+
+            httpConnectionOK.set(false);
+
+            // load configuration
+            FreeAtHomeBridgeHandlerConfiguration locConfig = getConfigAs(FreeAtHomeBridgeHandlerConfiguration.class);
+
+            ipAddress = locConfig.ipAddress;
+            password = locConfig.password;
+            username = locConfig.username;
+
+            // build base URL
+            baseUrl = "http://" + ipAddress + "/fhapi/v1/api";
+
+            // Open Http connection
+            if (!openHttpConnection()) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "@text/comm-error.http-wrongpass-or-ip");
+
+                thingReachable = false;
+            }
+
+            // Open the websocket connection for immediate status updates
+            if (!openWebSocketConnection()) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "@text/comm-error.not-able-open-websocketconnection");
+
+                thingReachable = false;
+            }
+
+            if (thingReachable) {
```

Comment:
If not, the binding will never recover? You could schedule a job for retrying establishment of the initial connection.

Reply:
The connection via HTTP and Websocket is not need to be retried, because the connection success is only depending on the following events:
* The WebSocket objects (Socket, SocketMonitor Thread) are allocated successfully --> No re-try shall be started
* The HTTP connection is successful to the SysAp is successful - The connection success is depending only on the User/Pass. If it is not defined correctly, the user shall change the pass and re-try with save. By pushing the SAVE button on the openHAB UI, then the Initialize method seems to be called again --> For my point of view no re-try is needed. It is handled by user interaction

Reply:
It's a matter of reliability. If the HTTP call would fail for any other reason (e.g. network issues), the binding would need manual user interaction to get online. For example, after a power outage, OH might be up running before other systems. See for example #11891 that gave me some headache in the past. I think that case is actually identical.

Reply:
I understand your point. But I will think about the use case which you mentioned. However in this case the opemHttpCo... method is just trying to open a connection to see that the user name is OK is not. if not from my point of view no automatic re-try is needed.

The use-case which you mentioned is that the openHAB is starting up faster then the free@home system. In this case the fault is fault is same as you will have if the IP address is wrong. No connection can be established.

I give you right that an automatic re-try is needed for the use-case when the openHAB is faster starting up but if you are not able to differentiate from the wrong IP address you are just continuously restarting even if the IP address is not valid. I think in this case also no automatic re-try is needed.

Automatic re-try can be implemented if the binding has a persistent storage, where the previous successful configuration can be stored with a success flag. It means the binding can decide to make automatic re-try even the connection cannot be established but the binding will know the connection shall be correct because the IP/User/Pass was in the past successful. Therefore the current connection issues shall be some network problems, then re-try until the network problem is solved.

I will look into it

Reply:
I won't insist, just wanted to let you that there is a corner case issue and that reliability can be improved. This kind of issue can cause pain in real life later. However, one comment:
> The use-case which you mentioned is that the openHAB is starting up faster then the free@home system. In this case the fault is fault is same as you will have if the IP address is wrong. No connection can be established.

Not entirely true. You are currently checking the HTTP status code, and considering 200 as "everything okay". Other status codes are considered the opposite. However, you are also catching multiple exceptions and treating them the same way. In some cases the inner exception will be `java.net.UnknownHostException` for the scenario mentioned, and in other cases you might see a timeout. In such cases it would make sense to retry. You could try to unplug the free@home system network cable and see what happens. Or you can choose to leave this issue for the future. 🙂

Reply:
I have still lots of ideas to improve. I would like to put this also in the backlog